posthog/api/dashboard.py

```diff
@@ -144,7 +144,7 @@ def get_queryset(self) -> QuerySet:
             raise NotFound()
         elif not self.request.user.is_authenticated or "team_id" not in self.get_parents_query_dict():
             raise AuthenticationFailed(detail="You're not logged in, but also not using add share_token.")
-
+        # TODO: Rework this to allow project-based permissioning restrictions
```

Comment:
We need to make sure this doesn't fall through the cracks, we want to avoid any potential data leaks.

Reply:
Good catch, added permissions + some tests